modules/my_health/app/controllers/my_health/v1/prescriptions_controller.rb

```diff
@@ -175,14 +175,20 @@ def filter_data_by_refill_and_renew(data)
         zero_date = Date.new(0, 1, 1)
 
         data.select do |item|
-          disp_status = item.disp_status
-          sorted_dispensed_date = item.sorted_dispensed_date.to_date
-
-          next true if ['Active', 'Active: Submitted'].include?(disp_status)
+          disp_status = item[:disp_status]
```

Comment:
To keep controllers clean, it seems that this code (amongst other methods) should be extracted outside of the controller potentially to a helper or even service object.

Reply:
@LindseySaari Sounds good, I will look into this. Thank you!